File: calcul/economie.py
def comparer_economie(dimensionnement, prix, conso_kwh_jour,
                       type_carburant, prix_carburant, source_carburant,
                       heures_groupe=8, inclut_entretien_groupe=True):
    """
    Compare le coût solaire vs groupe électrogène sur 10 ans.
    Nouvelle version avec carburant personnalisé et entretien détaillé.
    """

    # ─────────────────────────────────────────
    # DONNÉES GROUPE ÉLECTROGÈNE
    # ─────────────────────────────────────────

    # Consommation selon type carburant
    conso_litre_heure = {
        "essence": 0.8,   # litre/heure pour un groupe moyen
        "gasoil":  0.6    # gasoil plus économique
    }.get(type_carburant, 0.8)

    cout_carburant_jour    = heures_groupe * conso_litre_heure * prix_carburant
    cout_carburant_annuel  = cout_carburant_jour * 365

    # Entretien groupe électrogène détaillé
    entretien_groupe = {
        "vidange_huile":    {"freq_mois":3,  "cout":15000, "label":"Vidange huile moteur (tous les 3 mois)"},
        "filtre_air":       {"freq_mois":6,  "cout":8000,  "label":"Remplacement filtre à air (tous les 6 mois)"},
        "filtre_huile":     {"freq_mois":6,  "cout":5000,  "label":"Remplacement filtre à huile (tous les 6 mois)"},
        "bougie":           {"freq_mois":12, "cout":6000,  "label":"Remplacement bougies (essence uniquement · annuel)"},
        "revision_generale":{"freq_mois":12, "cout":35000, "label":"Révision générale annuelle"},
    }

    # Supprimer bougies si gasoil
    if type_carburant == "gasoil":
        del entretien_groupe["bougie"]

    # Calcul coût entretien annuel groupe
    cout_entretien_groupe_annuel = sum(
        round(e["cout"] * (12 / e["freq_mois"]))
        for e in entretien_groupe.values()
    )

    cout_initial_groupe  = 350000  # Prix groupe électrogène moyen
    cout_total_annuel_groupe = cout_carburant_annuel + cout_entretien_groupe_annuel

    # ─────────────────────────────────────────
    # DONNÉES SYSTÈME SOLAIRE
    # ─────────────────────────────────────────
    cout_initial_solaire = prix.get("total", 0)

    # Entretien annuel solaire (fixe)
    cout_entretien_solaire_annuel = 23000

    # Remplacement batterie
    duree_vie_bat = {"Lithium-Ion":12,"AGM":7,"Plomb-acide":5}
    type_bat      = dimensionnement["batterie"]["type"]
    duree_bat     = duree_vie_bat.get(type_bat, 7)
    prix_bat      = prix.get("batterie", 0)

    # ─────────────────────────────────────────
    # CALCUL SUR 10 ANS
    # ─────────────────────────────────────────
    total_groupe  = cout_initial_groupe
    total_solaire = cout_initial_solaire

    for annee in range(1, 11):
        total_groupe  += cout_total_annuel_groupe
        total_solaire += cout_entretien_solaire_annuel
        # Remplacement batterie
        if annee % duree_bat == 0 and annee < 10:
            total_solaire += prix_bat

    economies = total_groupe - total_solaire

    # Point de rentabilité
    cumul_groupe  = cout_initial_groupe
    cumul_solaire = cout_initial_solaire
    annee_rentabilite = None
    for annee in range(1, 11):
        cumul_groupe  += cout_total_annuel_groupe
        cumul_solaire += cout_entretien_solaire_annuel
        if annee % duree_bat == 0:
            cumul_solaire += prix_bat
        if cumul_solaire <= cumul_groupe and annee_rentabilite is None:
            annee_rentabilite = annee

    # Facture SBEE évitée (si hybride)
    # Estimation : consommation * tarif SBEE moyen (85 FCFA/kWh)
    tarif_sbee    = 85
    facture_sbee_evitee_mois = round(conso_kwh_jour * 30 * tarif_sbee)

    return {
        # Groupe
        "cout_initial_groupe":          cout_initial_groupe,
        "cout_carburant_annuel":        round(cout_carburant_annuel),
        "cout_entretien_groupe_annuel": cout_entretien_groupe_annuel,
        "cout_total_groupe_annuel":     round(cout_total_annuel_groupe),
        "cout_groupe_10ans":            round(total_groupe),
        "detail_entretien_groupe":      [
            {"label":e["label"],"cout_annuel":round(e["cout"]*(12/e["freq_mois"]))}
            for e in entretien_groupe.values()
        ],

        # Solaire
        "cout_initial_solaire":          cout_initial_solaire,
        "cout_entretien_solaire_annuel": cout_entretien_solaire_annuel,
        "cout_solaire_10ans":            round(total_solaire),
        "remplacement_batterie_an":      duree_bat,
        "cout_remplacement_batterie":    prix_bat,

        # Bilan
        "economies_10ans":       round(max(0, economies)),
        "annee_rentabilite":     annee_rentabilite or "10+",

        # SBEE
        "facture_sbee_evitee_mois": facture_sbee_evitee_mois,

        # Carburant
        "type_carburant":    type_carburant,
        "prix_carburant":    prix_carburant,
        "source_carburant":  source_carburant,
        "conso_litre_heure": conso_litre_heure,
        "cout_carburant_jour": round(cout_carburant_jour)
    }

File: static/js/economie.js
function selectCarburant(type) {
  etatCarburant.type = type;
  ["essence", "gasoil"].forEach((t) => {
    const btn = document.getElementById("fuel-" + t);
    if (btn) {
      btn.style.borderColor =
        t === type ? "var(--green-800)" : "var(--gray-200)";
      btn.style.background = t === type ? "var(--green-50)" : "#fff";
      btn.style.color = t === type ? "var(--green-800)" : "inherit";
    }
  });
}

function lancerEconomie() {
  if (!state.dimensionnement) {
    alert("Faites d'abord le dimensionnement.");
    return;
  }
  if (!document.getElementById("prix-carburant")) return;
  const prixCarburant =
    parseFloat(document.getElementById("prix-carburant").value) || 700;
  const heuresGroupe =
    parseFloat(document.getElementById("heures-groupe").value) || 8;
  showLoader("Calcul du bilan économique...");
  fetch("/economie", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      dimensionnement: state.dimensionnement,
      prix: state.prix,
      conso_kwh_jour: (state.consommationTotale || 0) / 1000,
      type_carburant: etatCarburant.type,
      prix_carburant: prixCarburant,
      source_carburant: "station",
      heures_groupe: heuresGroupe,
      inclut_entretien_groupe: true,
    }),
  })
    .then((r) => r.json())
    .then((data) => {
      hideLoader();
      if (data.succes) {
        state.economie = data.economie;
        afficherEconomie(data.economie);
      } else alert("Erreur : " + data.erreur);
    })
    .catch((e) => {
      hideLoader();
      alert("Erreur : " + e);
    });
}

function afficherEconomie(eco) {
  const ecoResults = document.getElementById("eco-results");
  if (!ecoResults) return;
  ecoResults.style.display = "block";
  const detGroupe = document.getElementById("eco-detail-groupe");
  if (!detGroupe) return;

  // Entretien groupe avec prix approximatifs
  const estEssence = etatCarburant.type === "essence";
  detGroupe.innerHTML = `
          <div style="display:flex;justify-content:space-between;font-size:12px;margin-bottom:4px;"><span style="color:#92400e;">Carburant (${eco.type_carburant})</span><span style="font-weight:600;color:#92400e;">${formatNombre(eco.cout_carburant_annuel || 0)} FCFA</span></div>
          <div style="border-top:1px dashed #fed7aa;margin:6px 0;padding-top:6px;">
            <div style="font-size:10px;color:#b45309;font-style:italic;margin-bottom:4px;">Entretien estimatif :</div>
            <div style="display:flex;justify-content:space-between;font-size:11px;margin-bottom:3px;color:#92400e;"><span>· Huile moteur (~1,5L · mensuel)</span><span>~3 000 FCFA/mois</span></div>
            <div style="display:flex;justify-content:space-between;font-size:11px;margin-bottom:3px;color:#92400e;"><span>· Filtre à air (tous les 3 mois)</span><span>~2 500 FCFA</span></div>
            <div style="display:flex;justify-content:space-between;font-size:11px;margin-bottom:3px;color:#92400e;"><span>· Filtre carburant (annuel)</span><span>~2 000 FCFA</span></div>
            ${estEssence ? `<div style="display:flex;justify-content:space-between;font-size:11px;margin-bottom:3px;color:#92400e;"><span>· Bougies (annuel)</span><span>~3 000 FCFA</span></div>` : ""}
            <div style="display:flex;justify-content:space-between;font-size:11px;margin-bottom:3px;color:#92400e;"><span>· Révision générale (annuel)</span><span>~15 000 FCFA</span></div>
            <div style="font-size:9px;color:#b45309;font-style:italic;margin-top:4px;">* Prix approximatifs — peuvent varier selon technicien et ville</div>
          </div>`;

  // Calendrier solaire avec échelle panneaux
  const nbPanneaux = state.dimensionnement
    ? state.dimensionnement.panneau.nombre
    : 1;
  let prixNettoyage, labelNettoyage;
  if (nbPanneaux <= 3) {
    prixNettoyage = "Gratuit / 1 000 FCFA si technicien sollicité";
    labelNettoyage = "1 à 3 panneaux";
  } else if (nbPanneaux <= 8) {
    prixNettoyage = "3 000 FCFA";
    labelNettoyage = "4 à 8 panneaux";
  } else if (nbPanneaux <= 15) {
    prixNettoyage = "5 000 FCFA";
    labelNettoyage = "9 à 15 panneaux";
  } else {
    prixNettoyage = "10 000 FCFA";
    labelNettoyage = "15+ panneaux";
  }

  document.getElementById("eco-detail-solaire").innerHTML = `
          <div style="display:flex;justify-content:space-between;font-size:12px;margin-bottom:4px;">
            <span style="color:var(--green-800);">Nettoyage panneaux <span style="font-size:10px;color:var(--gray-400);">(${labelNettoyage})</span></span>
            <span style="font-weight:600;color:var(--green-800);">${prixNettoyage}</span>
          </div>
          <div style="display:flex;justify-content:space-between;font-size:12px;margin-bottom:4px;">
            <span style="color:var(--green-800);">Vérification technique (annuel)</span>
            <span style="font-weight:600;color:var(--green-800);">À partir de 2 000 FCFA</span>
          </div>`;

  document.getElementById("eco-total-groupe-an").textContent =
    formatNombre(eco.cout_total_groupe_annuel || 0) + " FCFA";
  document.getElementById("eco-total-solaire-an").textContent =
    formatNombre(eco.cout_entretien_solaire_annuel || 0) + " FCFA";

  if (state.entretien) {
    const cal = document.getElementById("calendrier-entretien");
    const prixBat = state.prix ? state.prix.batterie || 0 : 0;
    const nbPan = state.dimensionnement
      ? state.dimensionnement.panneau.nombre
      : 1;
    let prixNett;
    if (nbPan <= 3) prixNett = "Gratuit / 1 000 FCFA si technicien sollicité";
    else if (nbPan <= 8) prixNett = "3 000 FCFA";
    else if (nbPan <= 15) prixNett = "5 000 FCFA";
    else prixNett = "10 000 FCFA";

    const typeBat = state.dimensionnement.batterie.type;
    const dureeVie = typeBat === "Lithium-Ion" ? 12 : typeBat === "AGM" ? 7 : 5;

    cal.innerHTML = `
            <div style="display:flex;align-items:flex-start;gap:10px;padding:10px;border-bottom:1px solid var(--gray-100);">
              <div style="width:8px;height:8px;border-radius:50%;margin-top:5px;flex-shrink:0;background:#166534;"></div>
              <div style="flex:1;">
                <div style="font-size:13px;font-weight:600;color:var(--gray-900);">Nettoyage des panneaux solaires</div>
                <div style="font-size:11px;color:var(--gray-500);">Hebdomadaire (harmattan) · Mensuel (saison des pluies)</div>
              </div>
              <div style="font-size:12px;font-weight:600;color:var(--green-800);">${prixNett}</div>
            </div>
            <div style="display:flex;align-items:flex-start;gap:10px;padding:10px;border-bottom:1px solid var(--gray-100);">
              <div style="width:8px;height:8px;border-radius:50%;margin-top:5px;flex-shrink:0;background:#f59e0b;"></div>
              <div style="flex:1;">
                <div style="font-size:13px;font-weight:600;color:var(--gray-900);">Vérification technique</div>
                <div style="font-size:11px;color:var(--gray-500);">Annuel · Connexions, câbles, régulateur</div>
              </div>
              <div style="text-align:right;">
                <div style="font-size:12px;font-weight:600;color:var(--green-800);">À partir de 2 000 FCFA</div>
              </div>
            </div>
            <div style="display:flex;align-items:flex-start;gap:10px;padding:10px;">
              <div style="width:8px;height:8px;border-radius:50%;margin-top:5px;flex-shrink:0;background:#dc2626;"></div>
              <div style="flex:1;">
                <div style="font-size:13px;font-weight:600;color:var(--gray-900);">Remplacement batterie ${typeBat}</div>
                <div style="font-size:11px;color:var(--gray-500);">Tous les ${dureeVie} ans</div>
              </div>
              <div style="text-align:right;">
                <div style="font-size:12px;font-weight:600;color:var(--orange-600);">${formatNombre(prixBat)} FCFA</div>
                <div style="font-size:10px;color:var(--gray-400);">+ main d'œuvre technicien</div>
              </div>
            </div>`;
  }

  document.getElementById("eco-init-groupe").textContent = "350.000 FCFA";
  document.getElementById("eco-init-solaire").textContent =
    formatNombre(eco.cout_initial_solaire || 0) + " FCFA";
  document.getElementById("eco-charges-groupe").textContent =
    formatNombre((eco.cout_total_groupe_annuel || 0) * 10) + " FCFA";
  document.getElementById("eco-charges-solaire").textContent = formatFCFA(
    (eco.cout_entretien_solaire_annuel || 0) * 10,
  );
  document.getElementById("eco-bat-replace").textContent =
    formatNombre(eco.cout_remplacement_batterie || 0) + " FCFA";
  document.getElementById("eco-total10-groupe").textContent =
    formatNombre(eco.cout_groupe_10ans || 0) + " FCFA";
  document.getElementById("eco-total10-solaire").textContent =
    formatNombre(eco.cout_solaire_10ans || 0) + " FCFA";
  document.getElementById("eco-economies").textContent =
    formatNombre(eco.economies_10ans || 0) + " FCFA";
  document.getElementById("eco-rentabilite").textContent =
    (eco.annee_rentabilite || "?") + " ans";
  if (state.typeSysteme === "hybrid" && eco.facture_sbee_evitee_mois) {
    document.getElementById("eco-sbee-box").style.display = "block";
    document.getElementById("eco-sbee").textContent =
      formatFCFA(eco.facture_sbee_evitee_mois) + " FCFA";
  }
  document.getElementById("eco-results").scrollIntoView({ behavior: "smooth" });
}

// ============================================================
// PAGE TECHNICIENS
// ============================================================
